app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java

```diff
@@ -1,8 +1,111 @@
  */
 
-public class UserProfileActivity {
+public class UserProfileActivity extends AppCompatActivity {
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_user_profile);
+        // Get the TextViews for the information that is going to be shown.
+        EditText userName = (EditText) findViewById(R.id.NameEditText);
+        EditText emailAddress = (EditText) findViewById(R.id.EmailEditText);
+        EditText phoneNumber = (EditText) findViewById(R.id.PhoneEditText);
+        // TODO Set the TextView to the proper values.
 }
+
+    public void editPhoneNumber(View v){
+        ImageButton cancelButton = (ImageButton) findViewById(R.id.CancelEditPhoneButton);
+        ImageButton saveButton = (ImageButton) findViewById(R.id.PhoneSaveEditButton);
+        ImageButton editButton = (ImageButton) findViewById(R.id.PhoneEditIconImageButton);
+        // Set visibility of the buttons.
+        editButton.setVisibility(View.INVISIBLE);
+        saveButton.setVisibility(View.VISIBLE);
+        cancelButton.setVisibility(View.VISIBLE);
+        TextView phoneNumber = (TextView) findViewById(R.id.PhoneEditText);
+        // Set it so the user can edit the TextView
+        phoneNumber.setFocusable(true);
+    }
+
+    public void saveEditedPhoneNumber(View v){
+        ImageButton cancelButton = (ImageButton) findViewById(R.id.CancelEditPhoneButton);
+        ImageButton saveButton = (ImageButton) findViewById(R.id.PhoneSaveEditButton);
+        ImageButton editButton = (ImageButton) findViewById(R.id.PhoneEditIconImageButton);
+        // Set visibility of the buttons.
+        editButton.setVisibility(View.VISIBLE);
+        saveButton.setVisibility(View.INVISIBLE);
+        cancelButton.setVisibility(View.INVISIBLE);
+        EditText phoneNumberText = (EditText) findViewById(R.id.PhoneEditText);
+        // Set it so the user can edit the TextView
+        phoneNumberText.setFocusable(false);
+        String phoneNumber = phoneNumberText.getText().toString();
+        // TODO Check if it is a valid phoneNumber
+        // TODO Update The information in elasticsearch
+    }
+    public void cancelEditPhoneNumber(View v){
+        ImageButton cancelButton = (ImageButton) findViewById(R.id.CancelEditPhoneButton);
+        ImageButton saveButton = (ImageButton) findViewById(R.id.PhoneSaveEditButton);
+        ImageButton editButton = (ImageButton) findViewById(R.id.PhoneEditIconImageButton);
+        // Set visibility of the buttons.
+        editButton.setVisibility(View.VISIBLE);
+        saveButton.setVisibility(View.INVISIBLE);
+        cancelButton.setVisibility(View.INVISIBLE);
+        EditText phoneNumberText = (EditText) findViewById(R.id.PhoneEditText);
+        // Set it so the user can edit the TextView
+        phoneNumberText.setFocusable(false);
+        // TODO Set the EditText to the value it originally was.
```

Comment:
I'm assuming this is just in case the user changes their mind and stops editing it right?